src/frontends/tensorflow/src/helper_transforms/block_lstm_replacer.cpp

```diff
@@ -0,0 +1,168 @@
+            seq_len_max,
+            make_shared<Constant>(ov::element::i64,
+                                  ov::Shape{1},
+                                  std::vector<int64_t>{block_lstm_node->get_attr_batch_size().get_length()}));
+
+        // prepare input data since LSTMSequence accept it in a format [batch_size, time_len, input_size]
+        auto x_order = make_shared<Constant>(ov::element::i64, ov::Shape{3}, std::vector<int64_t>{1, 0, 2});
+        auto x_adjusted = make_shared<Transpose>(x, x_order);
+
+        // create LSTMSequence node and reconnect inputs and normalized weights and bias
+        auto lstm_sequence = make_shared<LSTMSequence>(x_adjusted,
+                                                       init_hidden_state,
+                                                       init_cell_state,
+                                                       seq_len_max_adjusted,
+                                                       W,
+                                                       R,
+                                                       B,
+                                                       hidden_size.get_length(),
+                                                       LSTMSequence::direction::FORWARD);
+
+        // adjust output of concatenated of hidden states from LSTMSequence to have it in a format [time_len,
+        // batch_size, hidden_size]
+        // 1. squeeze extra dimension - num_directions
+        auto squeeze_axis = make_shared<Constant>(ov::element::i64, ov::Shape{1}, std::vector<int64_t>{1});
+        auto squeeze_output_hidden_states = make_shared<Squeeze>(lstm_sequence->output(0), squeeze_axis);
+        // 2. transpose the output to rotate batch and time dimensions
+        auto output_hidden_states_order =
+            make_shared<Constant>(ov::element::i64, ov::Shape{3}, std::vector<int64_t>{1, 0, 2});
+        auto output_hidden_states = make_shared<Transpose>(squeeze_output_hidden_states, output_hidden_states_order);
+
+        ov::NodeVector node_vector = {
+            new_weight_shape,  weight_reshape,  new_bias_shape,    bias_reshape,    reorder_const,   weights_axis,
+            weights_reorder,   bias_axis,       bias_reorder,      shift_const,     bias_split_lens, bias_split,
+            bias_first_shift,  bias_shift,      new_weight_shape2, weight_reshape2, transpose_order, new_bias_shape2,
+            weights_normalize, bias_normalized, WR_split_axis,     WR_split_lens,   WR_split,        lstm_sequence};
+        output_hidden_states->set_friendly_name(m.get_match_root()->get_friendly_name() + ":6");
+        ov::copy_runtime_info(block_lstm_node, node_vector);
+
+        // replace BlockLSTM with LSTMSequence manually instead of calling
+        // ov::replace_node(m.get_match_root(), lstm_sequence);
+        // because BlockLSTM has 7 outputs and LSTMSequence has three outputs
+        m.get_match_root()->output(6).replace(output_hidden_states->output(0));
```

Comment:
I'm not so sure that this is correct
I believe we need to get all target inputs of all outputs of BlockLSTM node and replace_source_output(...) for each of them

Reply:
I think this is correct because this is how it is done in `ov::replace_node`, for example.